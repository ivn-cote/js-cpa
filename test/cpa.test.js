const { findDuplicates, stringify } = require("../");
const path = require("path");
const fs = require("fs");

const fixturesDir = path.join(__dirname, "fixtures");

describe("fixtures", () => {
  fs.readdirSync(fixturesDir).forEach(name =>
    test(name, () => {
      const fixture = fs.readFileSync(path.join(fixturesDir, name)).toString();
      expect(
        stringify(findDuplicates(fixture, name), {
          colors: false,
          newline: false
        })
      ).toMatchSnapshot();
    })
  );
});
