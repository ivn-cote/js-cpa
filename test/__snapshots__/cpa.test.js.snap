// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fixtures match.js 1`] = `
"Found 2 matches

Duplicate 1

match.js → 5:5
function f() {}

match.js → 6:6
function aa() {}


Duplicate 2

match.js → 10:12
function test() {
  return function g1() {};
}

match.js → 20:22
function test1() {
  return function g2() {};
}

match.js → 25:27
function test2() {
  return function g3() {};
}

"
`;

exports[`fixtures nomatch.js 1`] = `
"Found 1 matches

Duplicate 1

nomatch.js → 2:2
function aaa() {}

nomatch.js → 8:8
function g2() {}

nomatch.js → 19:19
function f2() {}

"
`;
